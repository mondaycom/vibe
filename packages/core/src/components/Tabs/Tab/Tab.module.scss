@import "~monday-ui-style/dist/mixins";
@import "../../../styles/states";
@import "../../../styles/typography";

.tabWrapper {
  position: relative;
  display: inline-block;
  padding-left: 1px;
  padding-right: 1px;
  border-top: 1px solid transparent;
  border-bottom: 2px solid;
  border-bottom-color: var(--ui-background-color);
  color: var(--primary-text-color);
  text-align: center;
  height: 100%;
  margin: 0;
}

.tabWrapper:after {
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  content: "";
  border-bottom: solid 2px;
  border-bottom-color: var(--primary-color);
  transform: scaleX(0);
  z-index: 0;
}

.tabWrapper .tabInner {
  $margin-bottom: 1px;
  height: calc(100% - #{$margin-bottom});
  display: flex;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  padding: 4px 16px;
  @include vibe-text(text1, normal);
  user-select: none;
  margin-bottom: $margin-bottom;
  cursor: pointer;
  color: var(--primary-text-color);
}

.tabWrapper .tabInner:focus {
  outline: none;
}

.tabWrapper .tabInner .tabIcon {
  color: var(--icon-color);
  margin-right: var(--spacing-small);
}

.tabWrapper .tabInner .tabIcon.right {
  margin-right: 0;
  margin-left: var(--spacing-small);
}

.tabWrapper.tabFocusVisibleInset {
  @include focus-style-css-inset(3px, 3px);
  position: relative;
}

.tabWrapper.active:after {
  transform: scaleX(1);
  transition: transform var(--motion-productive-medium) var(--motion-timing-enter);
}

.tabWrapper.disabled .tabInner {
  color: var(--disabled-text-color);
  cursor: not-allowed;
}

.tabWrapper:not(.disabled) .tabInner:hover {
  border-radius: 4px;
  background-color: var(--primary-background-hover-color);
}
