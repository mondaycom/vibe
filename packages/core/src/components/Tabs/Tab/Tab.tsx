import cx from "classnames";
import React, { FC, forwardRef, ReactElement, useRef } from "react";
import { noop as NOOP } from "lodash-es";
import useMergeRef from "../../../hooks/useMergeRef";
import { getStyle } from "../../../helpers/typesciptCssModulesHelper";
import Icon from "../../Icon/Icon";
import VibeComponentProps from "../../../types/VibeComponentProps";
import { IconType } from "../../Icon";
import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import styles from "./Tab.module.scss";
import { SubIcon } from "../../../types/SubIcon";
import Tooltip, { TooltipProps } from "../../Tooltip/Tooltip";
import { ComponentVibeId } from "../../../tests/constants";
import { keyCodes } from "../../../constants";

export interface TabProps extends VibeComponentProps {
  /**
   * Class name applied to the inner tab content.
   */
  tabInnerClassName?: string;
  /**
   * The index value of the tab.
   */
  value?: number;
  /**
   * If true, disables the tab.
   */
  disabled?: boolean;
  /**
   * If true, marks the tab as active.
   */
  active?: boolean;
  /**
   * If true, applies focus styles to the tab.
   */
  focus?: boolean;
  /**
   * The icon displayed in the tab.
   */
  icon?: SubIcon;
  /**
   * The type of icon.
   */
  iconType?: IconType;
  /**
   * The position of the icon relative to the text.
   */
  iconSide?: string;
  /**
   * Callback fired when the tab is clicked.
   */
  onClick?: (value: number) => void;
  /**
   * Props passed to the tab's tooltip.
   */
  tooltipProps?: Partial<TooltipProps>;
  /**
   * The content displayed inside the tab.
   */
  children?: string | ReactElement | ReactElement[];
}

const Tab: FC<TabProps> = forwardRef(
  (
    {
      className,
      tabInnerClassName,
      id,
      value = 0,
      disabled = false,
      active = false,
      focus = false,
      onClick = NOOP,
      tooltipProps = {} as TooltipProps,
      icon,
      iconType,
      iconSide = "left",
      children,
      "data-testid": dataTestId
    }: TabProps,
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    function renderIconAndChildren() {
      if (!icon) return children;

      const iconElement = (
        <Icon
          ariaHidden={true}
          iconType={iconType}
          icon={icon}
          className={cx(styles.tabIcon, getStyle(styles, iconSide))}
          iconSize={18}
          ignoreFocusStyle
        />
      );

      const childrenArray = React.Children.toArray(children);

      if (iconSide === "left") {
        return [iconElement, ...childrenArray];
      }

      return [...childrenArray, iconElement];
    }

    function handleKeyDown(event: React.KeyboardEvent) {
      if (event.key === keyCodes.ENTER || event.key === keyCodes.SPACE) {
        event.preventDefault();
        !disabled && onClick(value);
      }
    }

    return (
      <Tooltip {...tooltipProps} content={tooltipProps.content}>
        <li
          ref={mergedRef}
          key={id}
          className={cx(styles.tabWrapper, className, {
            [styles.active]: active,
            [styles.disabled]: disabled,
            [styles.tabFocusVisibleInset]: focus
          })}
          id={id}
          role="tab"
          aria-selected={active}
          aria-disabled={disabled}
          data-testid={dataTestId || getTestId(ComponentDefaultTestId.TAB, id)}
          data-vibe={ComponentVibeId.TAB}
          onClick={() => !disabled && onClick(value)}
          onKeyDown={handleKeyDown}
        >
          <div className={cx(styles.tabInner, tabInnerClassName)}>{renderIconAndChildren()}</div>
        </li>
      </Tooltip>
    );
  }
);

export default Tab;
