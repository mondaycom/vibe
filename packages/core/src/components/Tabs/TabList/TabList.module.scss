@import "../../../styles/typography";

.tabsWrapper .tabsList {
  display: flex;
  flex-direction: row;
  list-style-type: none;
  font-size: 0;
  position: relative;
  height: 40px;
  outline: none;
  padding: 0;
}

.tabsWrapper .tabsList.sm {
  height: 32px;
}

.tabsWrapper .tabsList.lg {
  height: 48px;
}

.tabsWrapper.stretched .tabsList {
  width: 100%;
}

.tabsWrapper.stretched .tabsList .tabListTabWrapper {
  width: 100%;
}

.tabsWrapper.stretched .tabsList .tabListTabWrapper .tabListTabInner {
  width: 100%;
}

.tabsWrapper.disabled {
  opacity: 0;
  height: 0;
  padding-bottom: 5px;
  pointer-events: none;
}

.tabsWrapper.stretchedUnderline {
  border-bottom: 2px solid var(--ui-background-color);
}
