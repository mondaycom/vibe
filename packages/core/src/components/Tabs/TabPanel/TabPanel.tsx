import React, { FC, forwardRef, ReactElement, useRef } from "react";
import cx from "classnames";
import useMergeRef from "../../../hooks/useMergeRef";
import VibeComponentProps from "../../../types/VibeComponentProps";
import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import styles from "./TabPanel.module.scss";

export interface TabPanelProps extends VibeComponentProps {
  /**
   * The content inside the tab panel.
   */
  children?: ReactElement | ReactElement[] | string;
  /**
   * The index of the tab panel.
   */
  index?: number;
}

const TabPanel: FC<TabPanelProps> = forwardRef(({ className, id, children, index, "data-testid": dataTestId }, ref) => {
  const componentRef = useRef(null);
  const mergedRef = useMergeRef(ref, componentRef);

  return (
    <div
      key={`${id}_${index}`}
      ref={mergedRef}
      className={cx(styles.tabPanelWrapper, className)}
      id={id}
      data-testid={dataTestId || getTestId(ComponentDefaultTestId.TAB_PANEL, id)}
      role="tabpanel"
    >
      {children}
    </div>
  );
});

export default TabPanel;
