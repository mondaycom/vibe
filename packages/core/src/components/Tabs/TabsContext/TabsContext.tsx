import React, { FC, forwardRef, ReactElement, useCallback, useEffect, useRef, useState } from "react";
import useMergeRef from "../../../hooks/useMergeRef";
import usePrevious from "../../../hooks/usePrevious";
import VibeComponentProps from "../../../types/VibeComponentProps";
import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";

export interface TabsContextProps extends VibeComponentProps {
  /**
   * The index of the currently active tab.
   */
  activeTabId?: number;
  /**
   * The child elements representing the tab list and tab panels.
   */
  children?: ReactElement | ReactElement[];
}

type TabsChild = ReactElement & {
  type: Record<string, unknown>;
};

const TabsContext: FC<TabsContextProps> = forwardRef(
  ({ className, id, activeTabId = 0, children, "data-testid": dataTestId }, ref) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    const [previousActiveTabIdState, setPreviousActiveTabIdState] = useState(activeTabId);
    const [activeTabIdState, setActiveTabIdState] = useState(activeTabId);
    const prevActiveTabIdProp = usePrevious(activeTabId);

    useEffect(() => {
      // Update active tab if changed from props
      if (activeTabId !== prevActiveTabIdProp && activeTabId !== activeTabIdState) {
        setPreviousActiveTabIdState(activeTabIdState);
        setActiveTabIdState(activeTabId);
      }
    }, [activeTabId, activeTabIdState, prevActiveTabIdProp]);

    const onTabClick = useCallback(
      (tabId: number) => {
        setPreviousActiveTabIdState(activeTabIdState);
        setActiveTabIdState(tabId);
      },
      [activeTabIdState]
    );

    return (
      <div
        ref={mergedRef}
        className={className}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.TABS_CONTEXT, id)}
      >
        {React.Children.map(children, (child: TabsChild) => {
          if (child.type.isTabList) {
            const originalOnTabChange = child.props.onTabChange;
            const onTabChange = (tabId: number) => {
              onTabClick(tabId);
              originalOnTabChange?.(tabId);
            };
            return React.cloneElement(child, { activeTabId: activeTabIdState, onTabChange });
          }
          if (child.type.isTabPanels) {
            const animationDirection = previousActiveTabIdState < activeTabIdState ? "ltr" : "rtl";
            return React.cloneElement(child, { activeTabId: activeTabIdState, animationDirection });
          }
          return child;
        })}
      </div>
    );
  }
);

export default TabsContext;
