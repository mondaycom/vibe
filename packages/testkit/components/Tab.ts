import { Page, Locator } from "@playwright/test";
import { BaseElement } from "./BaseElement";

/**
 * Class representing a Tab element.
 * Extends the BaseElement class.
 */
export class Tab extends BaseElement {
  /**
   * Create a Tab.
   * @param {Page} page - The Playwright page object.
   * @param {Locator} locator - The locator for the Tab element.
   * @param {string} elementReportName - The name for reporting purposes.
   */
  constructor(page: Page, locator: Locator, elementReportName: string) {
    super(page, locator, elementReportName);
  }
}
