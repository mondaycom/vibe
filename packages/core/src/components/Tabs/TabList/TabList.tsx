import cx from "classnames";
import { camelCase } from "lodash-es";
import React, { FC, forwardRef, ReactElement, useCallback, useEffect, useMemo, useRef, useState } from "react";
import useGridKeyboardNavigation from "../../../hooks/useGridKeyboardNavigation/useGridKeyboardNavigation";
import useMergeRef from "../../../hooks/useMergeRef";
import usePrevious from "../../../hooks/usePrevious";
import VibeComponentProps from "../../../types/VibeComponentProps";
import { NOOP } from "../../../utils/function-utils";
import { TabProps } from "../Tab/Tab";
import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import { getStyle } from "../../../helpers/typesciptCssModulesHelper";
import styles from "./TabList.module.scss";

export interface TabListProps extends VibeComponentProps {
  onTabChange?: (tabId: number) => void;
  activeTabId?: number;
  tabType?: string;
  size?: string;
  children?: ReactElement<TabProps>[];
}

const TabList: FC<TabListProps> = forwardRef(
  (
    {
      className,
      id,
      onTabChange = NOOP,
      activeTabId = 0,
      tabType = "Compact",
      size,
      children,
      "data-testid": dataTestId
    },
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    const [activeTabState, setActiveTabState] = useState<number>(activeTabId);

    const prevActiveTabIdProp = usePrevious(activeTabId);

    useEffect(() => {
      // Update active tab if changed from props
      if (activeTabId !== prevActiveTabIdProp && activeTabId !== activeTabState) {
        setActiveTabState(activeTabId);
      }
    }, [activeTabId, prevActiveTabIdProp, activeTabState, setActiveTabState]);

    const disabledTabIds = useMemo(() => {
      const disabledIds = new Set<number>();
      React.Children.forEach(children, (child, index) => {
        if (child.props.disabled) {
          disabledIds.add(index);
        }
      });
      return disabledIds;
    }, [children]);

    const onTabSelect = useCallback(
      (tabId: number) => {
        if (disabledTabIds.has(tabId)) return;
        setActiveTabState(tabId);
        onTabChange && onTabChange(tabId);
      },
      [onTabChange, disabledTabIds]
    );

    const onTabClick = useCallback(
      (value: HTMLElement | void, tabId: number) => {
        const tabCallbackFunc = children[tabId].props?.onClick;
        if (disabledTabIds.has(tabId)) return;
        if (tabCallbackFunc) tabCallbackFunc(tabId);
        onTabSelect(tabId);
      },
      [children, disabledTabIds, onTabSelect]
    );

    const getItemByIndex = useCallback((index: number): ReactElement<TabProps> => children[index], [children]);
    const disabledIndexes = useMemo(() => Array.from(disabledTabIds), [disabledTabIds]);
    const ulRef = useRef();
    const { activeIndex: focusIndex, onSelectionAction } = useGridKeyboardNavigation({
      ref: ulRef,
      numberOfItemsInLine: children?.length,
      itemsCount: children?.length,
      getItemByIndex,
      onItemClicked: onTabClick,
      disabledIndexes
    });

    const tabsToRender = useMemo(() => {
      const childrenToRender = React.Children.map(children, (child, index) => {
        return React.cloneElement(child, {
          value: index,
          active: activeTabState === index,
          focus: focusIndex === index,
          onClick: onSelectionAction,
          className: cx(styles.tabListTabWrapper, child.props.className),
          tabInnerClassName: cx(styles.tabListTabInner, child.props.tabInnerClassName)
        });
      });
      return childrenToRender;
    }, [children, activeTabState, focusIndex, onSelectionAction]);

    return (
      <div
        ref={mergedRef}
        className={cx(styles.tabsWrapper, className, [getStyle(styles, camelCase(tabType))])}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.TAB_LIST, id)}
      >
        <ul ref={ulRef} tabIndex={0} className={cx(styles.tabsList, [getStyle(styles, size)])} role="tablist">
          {tabsToRender}
        </ul>
      </div>
    );
  }
);

Object.assign(TabList, {
  isTabList: true
});

export default TabList;
