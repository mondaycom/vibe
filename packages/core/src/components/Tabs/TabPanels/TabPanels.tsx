import cx from "classnames";
import { camelCase } from "lodash-es";
import React, { FC, forwardRef, ReactElement, useMemo, useRef } from "react";
import useMergeRef from "../../../hooks/useMergeRef";
import VibeComponentProps from "../../../types/VibeComponentProps";
import { TabPanelsAnimationDirection as TabPanelsAnimationDirectionEnum } from "./TabPanelsConstants";
import { TabPanelsAnimationDirection } from "./TabPanels.types";
import { TabPanelProps } from "../TabPanel/TabPanel";
import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import { getStyle } from "../../../helpers/typesciptCssModulesHelper";
import { withStaticProps } from "../../../types";
import styles from "./TabPanels.module.scss";

export interface TabPanelsProps extends VibeComponentProps {
  /**
   * The index of the currently active tab panel.
   */
  activeTabId?: number;
  /**
   * The animation direction when switching between tab panels.
   */
  animationDirection?: TabPanelsAnimationDirection;
  /**
   * The child elements representing tab panels.
   */
  children?: ReactElement<TabPanelProps> | ReactElement<TabPanelProps>[];
}

const TabPanels: FC<TabPanelsProps> & {
  animationDirections?: typeof TabPanelsAnimationDirectionEnum;
} = forwardRef(
  (
    { className, id, activeTabId = 0, animationDirection = "rtl", children, "data-testid": dataTestId }: TabPanelsProps,
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);
    const renderedTabs = useMemo(() => {
      return React.Children.map(children, (child, index) => {
        const isActiveTab = activeTabId === index;
        if (!isActiveTab) return null;
        const activeClass = isActiveTab ? "active" : "non-active";
        const animationClass = isActiveTab ? `animation-direction-${animationDirection}` : "";
        return React.cloneElement(child, {
          index,
          ...child.props,
          className: cx(
            styles.tabPanel,
            [getStyle(styles, activeClass)],
            [getStyle(styles, camelCase(animationClass))],
            child.props.className
          )
        });
      }).filter(Boolean);
    }, [children, activeTabId, animationDirection]);

    return (
      <div
        ref={mergedRef}
        className={cx(styles.tabPanelsWrapper, className)}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.TAB_PANELS, id)}
      >
        {renderedTabs}
      </div>
    );
  }
);

Object.assign(TabPanels, {
  isTabPanels: true
});

export default withStaticProps(TabPanels, {
  animationDirections: TabPanelsAnimationDirectionEnum
});
