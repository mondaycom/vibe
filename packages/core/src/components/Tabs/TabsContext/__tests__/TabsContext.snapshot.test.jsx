import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import TabsContext from "../TabsContext";
import TabPanel from "../../TabPanel/TabPanel";
import TabPanels from "../../TabPanels/TabPanels";
import Tab from "../../Tab/Tab";
import TabList from "../../TabList/TabList";

describe("TabsContext renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer
      .create(
        <TabsContext>
          <TabList>
            <Tab>First</Tab>
          </TabList>
          <TabPanels>
            <TabPanel>First slide</TabPanel>
          </TabPanels>
        </TabsContext>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with className", () => {
    const tree = renderer
      .create(
        <TabsContext className="test">
          <TabList>
            <Tab>First</Tab>
          </TabList>
          <TabPanels>
            <TabPanel>First slide</TabPanel>
          </TabPanels>
        </TabsContext>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with id", () => {
    const tree = renderer
      .create(
        <TabsContext id="test">
          <TabList>
            <Tab>First</Tab>
          </TabList>
          <TabPanels>
            <TabPanel>First slide</TabPanel>
          </TabPanels>
        </TabsContext>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with activeTabId", () => {
    const tree = renderer
      .create(
        <TabsContext activeTabId={1}>
          <TabList>
            <Tab>First</Tab>
            <Tab>Second</Tab>
          </TabList>
          <TabPanels>
            <TabPanel>First slide</TabPanel>
            <TabPanel>Second slide</TabPanel>
          </TabPanels>
        </TabsContext>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
