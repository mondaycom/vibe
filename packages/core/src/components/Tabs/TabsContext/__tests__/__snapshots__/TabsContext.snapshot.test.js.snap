// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsContext renders correctly with activeTabId 1`] = `
<div
  data-testid="tabs-context"
>
  <div
    className="tabsWrapper compact"
    data-testid="tab-list"
  >
    <ul
      className="tabsList"
      role="tablist"
      tabIndex={0}
    >
      <li
        aria-disabled={false}
        aria-selected={false}
        className="tabWrapper tabListTabWrapper"
        data-testid="tab"
        data-vibe="Tab"
        role="tab"
      >
        <a
          className="tabInner tabListTabInner"
          onClick={[Function]}
        >
          First
        </a>
      </li>
      <li
        aria-disabled={false}
        aria-selected={true}
        className="tabWrapper tabListTabWrapper active"
        data-testid="tab"
        data-vibe="Tab"
        role="tab"
      >
        <a
          className="tabInner tabListTabInner"
          onClick={[Function]}
        >
          Second
        </a>
      </li>
    </ul>
  </div>
  <div
    className="tabPanelsWrapper"
    data-testid="tab-panels"
  >
    <div
      className="tabPanelWrapper tabPanel active animationDirectionRtl"
      data-testid="tab-panel"
      role="tabpanel"
    >
      Second slide
    </div>
  </div>
</div>
`;

exports[`TabsContext renders correctly with className 1`] = `
<div
  className="test"
  data-testid="tabs-context"
>
  <div
    className="tabsWrapper compact"
    data-testid="tab-list"
  >
    <ul
      className="tabsList"
      role="tablist"
      tabIndex={0}
    >
      <li
        aria-disabled={false}
        aria-selected={true}
        className="tabWrapper tabListTabWrapper active"
        data-testid="tab"
        data-vibe="Tab"
        role="tab"
      >
        <a
          className="tabInner tabListTabInner"
          onClick={[Function]}
        >
          First
        </a>
      </li>
    </ul>
  </div>
  <div
    className="tabPanelsWrapper"
    data-testid="tab-panels"
  >
    <div
      className="tabPanelWrapper tabPanel active animationDirectionRtl"
      data-testid="tab-panel"
      role="tabpanel"
    >
      First slide
    </div>
  </div>
</div>
`;

exports[`TabsContext renders correctly with empty props 1`] = `
<div
  data-testid="tabs-context"
>
  <div
    className="tabsWrapper compact"
    data-testid="tab-list"
  >
    <ul
      className="tabsList"
      role="tablist"
      tabIndex={0}
    >
      <li
        aria-disabled={false}
        aria-selected={true}
        className="tabWrapper tabListTabWrapper active"
        data-testid="tab"
        data-vibe="Tab"
        role="tab"
      >
        <a
          className="tabInner tabListTabInner"
          onClick={[Function]}
        >
          First
        </a>
      </li>
    </ul>
  </div>
  <div
    className="tabPanelsWrapper"
    data-testid="tab-panels"
  >
    <div
      className="tabPanelWrapper tabPanel active animationDirectionRtl"
      data-testid="tab-panel"
      role="tabpanel"
    >
      First slide
    </div>
  </div>
</div>
`;

exports[`TabsContext renders correctly with id 1`] = `
<div
  data-testid="tabs-context_test"
  id="test"
>
  <div
    className="tabsWrapper compact"
    data-testid="tab-list"
  >
    <ul
      className="tabsList"
      role="tablist"
      tabIndex={0}
    >
      <li
        aria-disabled={false}
        aria-selected={true}
        className="tabWrapper tabListTabWrapper active"
        data-testid="tab"
        data-vibe="Tab"
        role="tab"
      >
        <a
          className="tabInner tabListTabInner"
          onClick={[Function]}
        >
          First
        </a>
      </li>
    </ul>
  </div>
  <div
    className="tabPanelsWrapper"
    data-testid="tab-panels"
  >
    <div
      className="tabPanelWrapper tabPanel active animationDirectionRtl"
      data-testid="tab-panel"
      role="tabpanel"
    >
      First slide
    </div>
  </div>
</div>
`;
